src/aws_encryption_sdk/streaming_client.py

```diff
@@ -209,6 +209,10 @@ def read(self, b=None):
         :returns: Processed (encrypted or decrypted) bytes from source stream
         :rtype: bytes
         """
+        # Any negative value for b is interpreted as a full read
```

Comment:
What is a full read?

Reply:
A full read means "read the entire message", and is triggered when either the number of bytes is not specified or when a negative value is specified.

https://docs.python.org/3/library/io.html#io.RawIOBase.read